File: SDKs/agco-ats.com/v1/python/src/sdk/models/operations/get_api_v2_vouchers_vouchercode_.py
from dataclasses import dataclass, field
from typing import Optional
from enum import Enum
from sdk.models import shared


@dataclass
class GetAPIV2VouchersVoucherCodePathParams:
    voucher_code: str = field(metadata={'path_param': { 'field_name': 'VoucherCode', 'style': 'simple', 'explode': False }})
    
class GetAPIV2VouchersVoucherCodeDeletedEnum(str, Enum):
    NOT_DELETED = "NotDeleted"
    DELETED = "Deleted"
    ALL = "All"


@dataclass
class GetAPIV2VouchersVoucherCodeQueryParams:
    deleted: Optional[GetAPIV2VouchersVoucherCodeDeletedEnum] = field(default=None, metadata={'query_param': { 'field_name': 'Deleted', 'style': 'form', 'explode': True }})
    

@dataclass
class GetAPIV2VouchersVoucherCodeRequest:
    path_params: GetAPIV2VouchersVoucherCodePathParams = field()
    query_params: GetAPIV2VouchersVoucherCodeQueryParams = field()
    

@dataclass
class GetAPIV2VouchersVoucherCodeResponse:
    content_type: str = field()
    status_code: int = field()
    body: Optional[bytes] = field(default=None)
    dealer_db_models_voucher: Optional[shared.DealerDbModelsVoucher] = field(default=None)
    

File: SDKs/agco-ats.com/v1/typescript/src/sdk/models/operations/getapiv2vouchersvouchercode.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetApiV2VouchersVoucherCodePathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=VoucherCode" })
  voucherCode: string;
}

export enum GetApiV2VouchersVoucherCodeDeletedEnum {
    NotDeleted = "NotDeleted",
    Deleted = "Deleted",
    All = "All"
}


export class GetApiV2VouchersVoucherCodeQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=Deleted" })
  deleted?: GetApiV2VouchersVoucherCodeDeletedEnum;
}


export class GetApiV2VouchersVoucherCodeRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetApiV2VouchersVoucherCodePathParams;

  @SpeakeasyMetadata()
  queryParams: GetApiV2VouchersVoucherCodeQueryParams;
}


export class GetApiV2VouchersVoucherCodeResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  body?: Uint8Array;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  dealerDbModelsVoucher?: shared.DealerDbModelsVoucher;

  @SpeakeasyMetadata()
  statusCode: number;
}
